src/core/desktopentry.cpp

```diff
@@ -178,11 +180,21 @@ ParsedDesktopEntryData DesktopEntry::parseText(const QString& id, const QString&
 
 		if (entries.contains(key)) {
 			const auto& old = entries.value(key);
+			const int oldScore = system.matchScore(old.first);
+			const int newScore = system.matchScore(locale);
 
-			auto oldScore = system.matchScore(old.first);
-			auto newScore = system.matchScore(locale);
+			bool shouldReplace = false;
+
+			if (newScore > oldScore) {
+				shouldReplace = true;
+			} else if (newScore == oldScore) {
+				// Prefer unlocalized (invalid locale) when both scores tie (usually 0)
+				if (!locale.isValid() && old.first.isValid()) {
+					shouldReplace = true;
+				}
+			}
 
-			if (newScore > oldScore || (oldScore == 0 && !locale.isValid())) {
+			if (shouldReplace) {
 				entries.insert(key, qMakePair(locale, value));
 			}
```

Comment:
What has meaningfully changed here? Returning zero in matchScore should always force fallback.

Reply:
I reverted this part, maybe not much just some explicitness - but its not relevant to the core bug fix